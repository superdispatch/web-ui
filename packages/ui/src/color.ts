//   - To update a color, modify the hex or rgba value associated with the desired color name.

export const Color = {
  White: "#fff",
  White50: "rgba(255, 255, 255, 0.5)",
  White40: "rgba(255, 255, 255, 0.4)",
  White20: "rgba(255, 255, 255, 0.2)",
  White10: "rgba(255, 255, 255, 0.1)",
  White08: "rgba(255, 255, 255, 0.08)",
  Transparent: "rgba(0, 0, 0, 0)",

  Black: "#000",
  Black50: "rgba(0, 0, 0, 0.5)",
  Black20: "rgba(0, 0, 0, 0.2)",

  //Colors with 30% opacity
  Dark30: "rgba(143, 148, 158, 0.3)",
  Blue30: "rgba(0, 112, 245, 0.3)",
  Green30: "rgba(3, 135, 47, 0.3)",
  Red30: "rgba(229, 35, 13, 0.3)",
  Silver30: "rgba(225, 229, 234, 0.3)",

  //Colors with 10% opacity
  Blue10: "rgba(0, 112, 245, 0.1)",
  Red10: "rgba(195, 25, 9, 0.1)",

  //
  // Neutral Colors
  //

  Dark50: "rgba(25, 35, 52, 0.5)",
  Dark100: "#8F949E",
  Dark300: "#5B6371",
  Dark500: "#192334",

  Silver200: "#F3F5F8",
  Silver400: "#E1E5EA",
  Silver500: "#C4CDD5",
  Silver400Alpha30: "rgba(225, 229, 234, 0.3)",

  //
  // Extended Palette
  //

  Blue50: "#EBF4FF",
  Blue300: "#0070F5",
  Blue500: "#0063DB",

  Green50: "#ECF9EF",
  Green300: "#03872F",
  Green500: "#007A29",

  Purple50: "#EFEEFC",
  Purple300: "#6559CF",
  Purple500: "#473ABB",

  Red50: "#FFEDEB",
  Red300: "#E5230D",
  Red300Aplha20: "rgba(229, 35, 13, 0.2)",
  Red500: "#C31909",

  Teal50: "#E3F6FC",
  Teal300: "#007EAB",
  Teal500: "#00678A",

  Yellow50: "#FFF7DC",
  Yellow300: "#E8671C",
  Yellow500: "#B84807",
} as const;

// `ColorDark`: Represents the color palette for dark themes.
//     - Tailored for dark theme usage.
//     - Update colors by modifying the associated values.

export const ColorDark = {
  White: "#151B22",
  White50: "rgba(21, 27, 34, 0.5)",
  White40: "rgba(21, 27, 34, 0.4)",
  White20: "rgba(21, 27, 34, 0.2)",
  White10: "rgba(21, 27, 34, 0.1)",
  White08: "rgba(21, 27, 34, 0.08)",
  Transparent: "rgba(0, 0, 0, 0)",

  Black: "#000",
  Black50: "rgba(0, 0, 0, 0.5)",
  Black20: "rgba(0, 0, 0, 0.2)",

  //Colors with 30% opacity
  Dark30: "rgba(143, 148, 158, 0.3)",
  Blue30: "rgba(0, 112, 245, 0.3)",
  Green30: "rgba(3, 135, 47, 0.3)",
  Red30: "rgba(229, 35, 13, 0.3)",
  Silver30: "rgba(225, 229, 234, 0.3)",

  //Colors with 10% opacity
  Blue10: "rgba(0, 112, 245, 0.1)",
  Red10: "rgba(195, 25, 9, 0.1)",

  //
  // Neutral Colors
  //

  Dark50: "rgba(25, 35, 52, 0.5)",
  Dark100: "#6E7A8A",
  Dark300: "#AEB4BD",
  Dark500: "#FFFFFF",

  Silver200: "#0D1017",
  Silver400: "#30373D",
  Silver500: "#47505B",
  Silver400Alpha30: "rgba(48, 55, 61, 0.3)",

  //
  // Extended Palette
  //

  Blue50: "#0E2947",
  Blue300: "#0070F5",
  Blue500: "#55ADFF",

  Green50: "#092E16",
  Green300: "#03872F",
  Green500: "#5EE18A",

  Purple50: "#262247",
  Purple300: "#6559CF",
  Purple500: "#BFA0FF",

  Red50: "#3D0A06",
  Red300: "#E5230D",
  Red300Aplha20: "rgba(233, 27, 12, 0.3)",
  Red500: "#FF6359",

  Teal50: "#11333D",
  Teal300: "#007EAB",
  Teal500: "#31C7F5",

  Yellow50: "#33200A",
  Yellow300: "#E8671C",
  Yellow500: "#FCA542",
} as const;

//   - Designed for more dynamic theming using CSS custom properties.
//   - Update colors by changing the CSS variable references.

export const ColorDynamic = {
  White: "var(--sd-white)",
  White50: "var(--sd-white-50)",
  White40: "var(--sd-white-40)",
  White20: "var(--sd-white-20)",
  White10: "var(--sd-white-10)",
  White08: "var(--sd-white-08)",
  Transparent: "var(--sd-transparent)",

  Black: "var(--sd-black)",
  Black50: "var(--sd-black-50)",
  Black20: "var(--sd-black-20)",

  // Colors with 30% opacity
  Dark30: "var(--sd-dark-30)",
  Blue30: "var(--sd-blue-30)",
  Green30: "var(--sd-green-30)",
  Red30: "var(--sd-red-30)",
  Silver30: "var(--sd-silver-30)",

  // Colors with 10% opacity
  Blue10: "var(--sd-blue-10)",
  Red10: "var(--sd-red-10)",

  //
  // Neutral Colors
  //

  Dark50: "var(--sd-dark-50)",
  Dark100: "var(--sd-dark-100)",
  Dark300: "var(--sd-dark-300)",
  Dark500: "var(--sd-dark-500)",

  Silver200: "var(--sd-silver-200)",
  Silver400: "var(--sd-silver-400)",
  Silver500: "var(--sd-silver-500)",
  Silver400Alpha30: "var(--sd-silver-400-alpha-30)",

  //
  // Extended Palette
  //

  Blue50: "var(--sd-blue-50)",
  Blue300: "var(--sd-blue-300)",
  Blue500: "var(--sd-blue-500)",

  Green50: "var(--sd-green-50)",
  Green300: "var(--sd-green-300)",
  Green500: "var(--sd-green-500)",

  Purple50: "var(--sd-purple-50)",
  Purple300: "var(--sd-purple-300)",
  Purple500: "var(--sd-purple-500)",

  Red50: "var(--sd-red-50)",
  Red300: "var(--sd-red-300)",
  Red500: "var(--sd-red-500)",
  Red300Aplha20: "var(--sd-red-300-alpha-20)",

  Teal50: "var(--sd-teal-50)",
  Teal300: "var(--sd-teal-300)",
  Teal500: "var(--sd-teal-500)",

  Yellow50: "var(--sd-yellow-50)",
  Yellow300: "var(--sd-yellow-300)",
  Yellow500: "var(--sd-yellow-500)",
} as const;
