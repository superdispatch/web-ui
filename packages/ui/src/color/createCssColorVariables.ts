import { Color, ColorDark } from "../color";

const CSS_VAR_PREFIX = "sd";

function colorKeyToCssVarName(key: string) {
  return key.toLowerCase().split(/(\d+)/).filter(Boolean).join("-");
}

export function createCssColorVariables(colorScheme: "light" | "dark") {
  const color = colorScheme === "light" ? Color : ColorDark;

  return Object.entries(color).reduce(
    (acc, [key, value]) => {
      acc[`--${CSS_VAR_PREFIX}-${colorKeyToCssVarName(key)}`] = value;
      return acc;
    },
    {} as Record<string, string>,
  );
}
