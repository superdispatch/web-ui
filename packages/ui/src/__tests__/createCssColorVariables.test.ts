import { describe, it, expect } from "vitest";
import { createCssColorVariables } from "../color/createCssColorVariables";

describe("createCssColorVariables", () => {
  it("should create css variables for light mode", () => {
    const variables = createCssColorVariables("light");
    expect(variables).toMatchSnapshot();
  });
  it("should create css variables for dark mode", () => {
    const variables = createCssColorVariables("dark");
    expect(variables).toMatchSnapshot();
  });
});
