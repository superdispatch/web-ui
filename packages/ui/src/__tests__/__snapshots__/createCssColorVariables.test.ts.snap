// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createCssColorVariables > should create css variables for dark mode 1`] = `
{
  "--sd-black": "#000",
  "--sd-black-20": "rgba(0, 0, 0, 0.2)",
  "--sd-black-50": "rgba(0, 0, 0, 0.5)",
  "--sd-blue-10": "rgba(0, 112, 245, 0.1)",
  "--sd-blue-30": "rgba(0, 112, 245, 0.3)",
  "--sd-blue-300": "#0070F5",
  "--sd-blue-50": "#0E2947",
  "--sd-blue-500": "#55ADFF",
  "--sd-dark-100": "#6E7A8A",
  "--sd-dark-30": "rgba(143, 148, 158, 0.3)",
  "--sd-dark-300": "#AEB4BD",
  "--sd-dark-50": "rgba(25, 35, 52, 0.5)",
  "--sd-dark-500": "#FFFFFF",
  "--sd-green-30": "rgba(3, 135, 47, 0.3)",
  "--sd-green-300": "#03872F",
  "--sd-green-50": "#092E16",
  "--sd-green-500": "#5EE18A",
  "--sd-purple-300": "#6559CF",
  "--sd-purple-50": "#262247",
  "--sd-purple-500": "#BFA0FF",
  "--sd-red-10": "rgba(195, 25, 9, 0.1)",
  "--sd-red-30": "rgba(229, 35, 13, 0.3)",
  "--sd-red-300": "#E5230D",
  "--sd-red-300-aplha-20": "rgba(233, 27, 12, 0.3)",
  "--sd-red-50": "#3D0A06",
  "--sd-red-500": "#FF6359",
  "--sd-silver-200": "#0D1017",
  "--sd-silver-30": "rgba(225, 229, 234, 0.3)",
  "--sd-silver-400": "#30373D",
  "--sd-silver-400-alpha-30": "rgba(48, 55, 61, 0.3)",
  "--sd-silver-500": "#47505B",
  "--sd-teal-300": "#007EAB",
  "--sd-teal-50": "#11333D",
  "--sd-teal-500": "#31C7F5",
  "--sd-transparent": "rgba(0, 0, 0, 0)",
  "--sd-white": "#151B22",
  "--sd-white-08": "rgba(21, 27, 34, 0.08)",
  "--sd-white-10": "rgba(21, 27, 34, 0.1)",
  "--sd-white-20": "rgba(21, 27, 34, 0.2)",
  "--sd-white-40": "rgba(21, 27, 34, 0.4)",
  "--sd-white-50": "rgba(21, 27, 34, 0.5)",
  "--sd-yellow-300": "#E8671C",
  "--sd-yellow-50": "#33200A",
  "--sd-yellow-500": "#FCA542",
}
`;

exports[`createCssColorVariables > should create css variables for light mode 1`] = `
{
  "--sd-black": "#000",
  "--sd-black-20": "rgba(0, 0, 0, 0.2)",
  "--sd-black-50": "rgba(0, 0, 0, 0.5)",
  "--sd-blue-10": "rgba(0, 112, 245, 0.1)",
  "--sd-blue-30": "rgba(0, 112, 245, 0.3)",
  "--sd-blue-300": "#0070F5",
  "--sd-blue-50": "#EBF4FF",
  "--sd-blue-500": "#0063DB",
  "--sd-dark-100": "#8F949E",
  "--sd-dark-30": "rgba(143, 148, 158, 0.3)",
  "--sd-dark-300": "#5B6371",
  "--sd-dark-50": "rgba(25, 35, 52, 0.5)",
  "--sd-dark-500": "#192334",
  "--sd-green-30": "rgba(3, 135, 47, 0.3)",
  "--sd-green-300": "#03872F",
  "--sd-green-50": "#ECF9EF",
  "--sd-green-500": "#007A29",
  "--sd-purple-300": "#6559CF",
  "--sd-purple-50": "#EFEEFC",
  "--sd-purple-500": "#473ABB",
  "--sd-red-10": "rgba(195, 25, 9, 0.1)",
  "--sd-red-30": "rgba(229, 35, 13, 0.3)",
  "--sd-red-300": "#E5230D",
  "--sd-red-300-aplha-20": "rgba(229, 35, 13, 0.2)",
  "--sd-red-50": "#FFEDEB",
  "--sd-red-500": "#C31909",
  "--sd-silver-200": "#F3F5F8",
  "--sd-silver-30": "rgba(225, 229, 234, 0.3)",
  "--sd-silver-400": "#E1E5EA",
  "--sd-silver-400-alpha-30": "rgba(225, 229, 234, 0.3)",
  "--sd-silver-500": "#C4CDD5",
  "--sd-teal-300": "#007EAB",
  "--sd-teal-50": "#E3F6FC",
  "--sd-teal-500": "#00678A",
  "--sd-transparent": "rgba(0, 0, 0, 0)",
  "--sd-white": "#fff",
  "--sd-white-08": "rgba(255, 255, 255, 0.08)",
  "--sd-white-10": "rgba(255, 255, 255, 0.1)",
  "--sd-white-20": "rgba(255, 255, 255, 0.2)",
  "--sd-white-40": "rgba(255, 255, 255, 0.4)",
  "--sd-white-50": "rgba(255, 255, 255, 0.5)",
  "--sd-yellow-300": "#E8671C",
  "--sd-yellow-50": "#FFF7DC",
  "--sd-yellow-500": "#B84807",
}
`;
